src/web3/KeychainWallet.ts

```diff
@@ -19,14 +25,52 @@ export class KeychainWallet extends RemoteWallet<KeychainSigner> implements Unlo
     super()
   }
 
-  async loadAccountSigners(): Promise<Map<string, KeychainSigner>> {
-    const accounts = await listStoredAccounts(this.importMnemonicAccount)
-    const addressToSigner = new Map<string, KeychainSigner>()
+  /**
+   * Gets a list of addresses that have been registered
+   */
+  getAccounts(): Address[] {
+    return Array.from(this.keychainAccounts.keys())
+  }
+
+  /**
+   * Gets a single account been registered
+   */
+  getAccount(address: string): KeychainAccountManager {
+    const normalizedAddress = normalizeAddressWith0x(address)
+    if (!this.keychainAccounts.has(normalizedAddress)) {
+      throw new Error(`Could not find address ${normalizedAddress}`)
+    }
+    return this.keychainAccounts.get(normalizedAddress)!
+  }
 
+  /**
+   * Returns true if account has been registered
+   * @param address Account to check
+   */
+  hasAccount(address?: Address): boolean {
+    if (address) {
+      const normalizedAddress = normalizeAddressWith0x(address)
+      return this.keychainAccounts.has(normalizedAddress)
+    } else {
+      return false
+    }
+  }
+
+  protected getSigner(address: string): ContractKitSigner {
+    const normalizedAddress = normalizeAddressWith0x(address)
+    if (!this.keychainAccounts.has(normalizedAddress)) {
+      throw new Error(`Could not find address ${normalizedAddress}`)
+    }
+    return this.keychainAccounts.get(normalizedAddress)!.unlockedContractKitSigner
+  }
+
+  async loadAccountSigners(): Promise<Map<string, ContractKitSigner>> {
+    const accounts = await listStoredAccounts(this.importMnemonicAccount)
     accounts.forEach((account) => {
-      addressToSigner.set(account.address, new KeychainSigner(account))
+      const accountManager = new KeychainAccountManager(account)
+      this.keychainAccounts.set(account.address, accountManager)
     })
-    return addressToSigner
+    return new Map<string, ContractKitSigner>()
```

Comment:
It doesn't actually matter what this returns anymore. This function has to be implemented since its an abstract method on RemoteWallet but since we overrode `getSigner` to depend on this.keychainAccounts it doesn't matter what we set the signers to. I can add a comment for that

Reply:
still seems preferable to give a meaningful return value.. otherwise it's a footgun